src/datepicker/styled-components.js

```diff
@@ -184,50 +204,46 @@ function calculateBorderRadius(props): ?BorderRadiusT {
     $hasRangeSelected,
     $theme: {borders},
   } = props;
-  if (borders.useRoundedCorners) {
-    if ($selected) {
-      if (!$range) {
-        return getBorderRadius(borders.radius200, borders.radius200);
+  const radius = borders.datepickerInteractionBorderRadius;
+
+  if ($selected) {
+    if (!$range) {
+      return getBorderRadius(radius, radius);
+    } else {
+      if ($hasRangeSelected) {
+        return $startDate
+          ? getBorderRadius(radius, 0)
+          : getBorderRadius(0, radius);
       } else {
-        if ($hasRangeSelected) {
-          return $startDate
-            ? getBorderRadius(borders.radius200, 0)
-            : getBorderRadius(0, borders.radius200);
+        if ($hasRangeHighlighted) {
+          return $hasRangeOnRight
+            ? getBorderRadius(radius, 0)
```

Comment:
This should not be the same as `datepickerInteractionBorderRadius`. In case we want to provide easy customization of the selected date corners it should be decoupled from the main look border radius. I'd rather not provide a theme override for the selected date corners radius considering we'll be updating it furthermore accordingly to the latest design.

Reply:
We can hardcode these (current), make another variable for them (`datepickerRangeBorderRadius`), or share the `datepickerInteractionBorderRadius` (which I think is equivalent to the current corner styling). Which one do you like?

Reply:
What do you think about leaving this part as is for now using `borders.radius200`?

Reply:
I prefer a new variable - It seems like we will need one eventually (assuming we can't couple this with the `datepickerInteractionBorderRadius`) and I'm not sure why we should wait. Will the overall `datepicker` design change so greatly the variable no longer makes sense?

Once it is time to square things off we can just change the default theme value and then people can opt out of those changes by tweaking their theme.

Reply:
we should be judicious when exposing theme knobs to consumers. what is the core issue this pr is meant to fix? i see it as making it simple to set 'surfaces' and 'inputs' to have rounded corners (button, card, input, menu, toast). the rest does not really matter, and in the worst situation others can be easily added. to me, radii theme values other than the ones i enumerated above should be removed

Reply:
Look at the WIP datepicker in Figma. We didn't let the selected date to be modified by setting the `useRoundedCorners` and it's not something directly depends on rounded corners usage in UI.

Reply:
How does this look then?
```
buttonBorderRadius,
inputBorderRadius,
cardBorderRadius,
menuBorderRadius,
popoverBorderRadius  // Do we need both popover and toast?
```
This means some corners will be hard coded- the developer will need to use `override` to customize.

Reply:
this lgtm - agreed on the popover only